src/internal/TouchRipple.js

```diff
@@ -179,7 +179,10 @@ class TouchRipple extends Component {
     style.height = `${rippleSize}px`;
     style.width = `${rippleSize}px`;
     style.top = `${top}px`;
-    style.left = `${left}px`;
+    if (this.context.muiTheme.isRtl) {
+      style.left = 'auto'; style.right = `${left}px`;
```

Comment:
I don't think that we should fix it like this.
This line is making the `style` `isRTL` independent.
That's what we are looking for, but the implementation is really tricky 🙈 .
We are using the `right` property so it can be reverted to the `left` one later on.

What about adding an option so the `rtl.js` can ignore some object, this one is a perfect example.
Then, using this option here.